src/components/TodoApp.tsx

```diff
@@ -0,0 +1,249 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, {
+  useState,
+  useContext,
+  useMemo,
+  useEffect,
+} from 'react';
+import classNames from 'classnames';
+import { FilterChoise } from '../types/FilterChoise';
+import { TodosContext } from '../context/TodoContext';
+import { TodoList } from './TodoList';
+import { TodoFilter } from './TodoFilter';
+import { ErrorNotification } from './ErrorNotification';
+import { addTodo, deleteTodo, updateTodo } from '../api/todos';
+import { USER_ID } from '../utils/constans';
+import { Todo } from '../types/Todo';
+
+export const TodoApp: React.FC = () => {
+  const [filterChoise, setFilterChoise] = useState(FilterChoise.All);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [toHideError, setToHideError] = useState(true);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [processedTodoIds, setProcessedTodoIds] = useState<number[]>([]);
+
+  const [todos, setTodos, errorMsg, setErrorMsg] = useContext(TodosContext);
+
+  // eslint-disable-next-line no-lone-blocks
+  { // for next tasks, i have problem with linter without it
+  }
+
+  const filteredTodos = useMemo(() => todos.filter((todo) => {
+    switch (filterChoise) {
+      case FilterChoise.Active:
+        return !todo.completed;
+      case FilterChoise.Completed:
+        return todo.completed;
+      default:
+        return true;
+    }
+  }), [todos, filterChoise]);
+
+  const uncompleatedTodos = todos.filter((todo) => !todo.completed);
+  const completedTodos = todos.filter((todo) => todo.completed);
+  const uncompleatedTodosText
+    = uncompleatedTodos.length === 1 ? 'item' : 'items';
+
+  const [allCompleated, setAllCompleated] = useState(
+    uncompleatedTodos.length === 0,
+  );
+
+  useMemo(() => {
+    setAllCompleated(uncompleatedTodos.length === 0);
+  }, [uncompleatedTodos.length]);
```

Comment:
just use `useMemo` without state variable
```suggestion
  const allCompleated = useMemo(() => uncompleatedTodos.length === 0, [uncompleatedTodos.length]);
```